src/tox/session/env_select.py

```diff
@@ -389,6 +391,16 @@ def _mark_provision(self, on: bool, provision_tox_env: str) -> None:  # noqa: FB
         self._provision = on, provision_tox_env
 
 
+def _find_env_match(value: str, state_conf: set[str]) -> bool:
+    return any(value in conf.split("-") for conf in state_conf) or any(
```

Comment:
I'd like to add some comments to explain what is happening here, but I'm not familiar with the correct nomenclature.

Reply:
tox environment names are made up of factors (so `py39-magic-test` is a tox environment made up of three factors `py39`, `magic`, `test`).

Reply:
There's some more of these here https://github.com/tox-dev/tox/blob/main/src/tox/config/loader/ini/factor.py and explanation how those are split/generated 👍